solutions/devsprint-riccardo-1/OnboardingChallenge/Components/ButtonView.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  ButtonView.swift
+//  OnboardingChallenge
+//
+//  Created by Pedro Gabriel on 15/12/21.
+//
+
+import UIKit
+
+struct ButtonViewConfiguration {
+    let title: String
+    var pressedButton: (() -> Void)?
+}
+
+class ButtonView: UIView, ViewCode {
+    // MARK: Properties
+    private var pressedButton: (() -> Void)?
+    
+    private lazy var button: UIButton = {
+        var button = UIButton(type: .system)
+        
+        button.layer.cornerRadius = 15
+        button.backgroundColor = .systemBlue
+        button.titleLabel?.tintColor = .white
+        button.titleLabel?.font = .systemFont(ofSize: 21, weight: .semibold)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        
+        button.addTarget(self, action: #selector(buttonPressed), for: .touchUpInside)
+        return button
+    }()
+    
+    // MARK: Initializers
+    init() {
+        super.init(frame: .zero)
+        setupViews()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: Overrides
+    override var intrinsicContentSize: CGSize {
+        return CGSize(width: UIView.noIntrinsicMetric, height: 65)
```

Comment:
Great idea to leverage the `intrinsicContentSize`. Given that we are encapsulating a button, shouldn't we return the `button.instrinsicContentSize`?

Reply:
This is the only comment standing.